lib/bridge/MatrixHandler.js

```diff
@@ -468,11 +468,53 @@ MatrixHandler.prototype._onAdminMessage = Promise.coroutine(function*(req, event
         return;
     }
     else if (cmd === "!quit") {
-        // clientList.filter(
-        //     (bridgedClient) => bridgedClient.server.domain === ircServer.domain
-        // ).forEach((bridgedClient) => {
-        //     bridgedClient.kill();
-        // });
+        // Filter to get the clients for the [specified] server
+        let clients = clientList.filter(
+            (bridgedClient) => bridgedClient.server.domain === ircServer.domain
+        );
+        if (clients.length === 0) {
```

Comment:
We should never have `> 1` client in this list. We should check for that and then just grab the first entry so we don't need to do loops over this 1-element list.